### ISSUE TITLE
--sync-from-start

### ISSUE THREAD

**Issue by Testato** [vysheng/tg#834]
what do this argument ?

i used telegram cli many month ago, now i restarted it and on the consolle appear, slowly, all the message that i sended/received (i use the same username on Telegram CLI and on my Android Telegram)

But i do not need that all message is fetched, because until it not finish to fetch i cannot use Telegram CLI.

Is (--sync-from-start) an option for tell to Telegram CLI of NOT download the past message ?

**Comment by LukeLR** [comment]
No, as far as I know, this argument is used to tell tg-cli to download all the messages in your account, even those that were sent before your instance of tg-cli was installed.

**Comment by Testato** [comment]
So the retrieve of all message from last login is a standard behavior ?

If it is normal it is very slow, i need one complete day for retrieve 3
month of message.

I do not need the message history on tg-cli so is there an argument for
stop it ?

Thanks fo the answer
Il 26/ott/2015 19:34, "Lukas Rose" <notifications@github.com> ha scritto:

> No, as far as I know, this argument is used to tell tg-cli to download all
> the messages in your account, even those that were sent before your
> instance of tg-cli was installed.
>
>
>

**Comment by Testato** [comment]
Maybe --deamonize not only block the visualization of the new message, but also stop the history retrieve ?

**Comment by LukeLR** [comment]
Yes, this is Standard behaviour of tg-cli, and there's nothing bad about it, it's useful to have a steam of past messages. Usually, you'd create a second account for your instance of tg-cli, that you're not using for your communication.

But anyways, it shouldn't take that long for synchronisation. Is your CPU or network connection that slow?

**Comment by Testato** [comment]
I use it on an embedded board, like a raspberry but different.
For this kind of use will be very usefull write a new option to pass by an
argument for blocking the retrive of the message. Sometingh like:
--no_message_history
Il 27/ott/2015 11:05, "Lukas Rose" <notifications@github.com> ha scritto:

> Yes, this is Standard behaviour of tg-cli, and there's nothing bad about
> it, it's useful to have a steam of past messages. Usually, you'd create a
> second account for your instance of tg-cli, that you're not using for your
> communication.
>
> But anyways, it shouldn't take that long for synchronisation. Is your CPU
> or network connection that slow?
>
>

**Comment by LukeLR** [comment]
Yes, but I don't understand, why syncing takes so incredibly long for
you. I'm developing on a Raspberry as well, and even when it has to
fetch thousands of past messages, let's say 3.000, the process finishes
after a few minutes. So there must be definitely another issue.

**Comment by Testato** [comment]
May be the Lua parsing ?
I use the lua example script, integrated in the tg-cli, for parse all
message (if i found a specific command i make something).
Maybe this continous parsing make all slow ?

A question ? All the message thst appear in the terminal is stored
somewhare ?
Maybe the terminal shell have a limit on the number of characthers that can
appear ?
Il 30/ott/2015 12:38, "Lukas Rose" <notifications@github.com> ha scritto:

**Comment by LukeLR** [comment]
Yes, of course, having lua parsing enabled for *each* message will cause an *extreme slowdown* on processing these messages! You should really implement an option to skip past messages when the client reconnects, check the lua-manual for:

on_binlog_replay_end() - it is called when replay of old events end. Any updates prior this call were already received by the client some time ago.

You should add a Boolean variable to your script, that's set to "false" by default. Add an if-statement to your processing method that skips all messages while the flag is false. Only after the method "on_binlog_replay_end()" is called, the flag is set to true, and any following messages will be parsed.

The terminal buffer is usually not saved, but this depends on your terminal emulator. If you're just connecting to your system by telnet/ssh, this should not be the case.

**Comment by Testato** [comment]
thanks for the tip, tomorrow i will try it.
but i found this issue that is about on_binlog_replay_end
https://github.com/vysheng/tg/issues/497

Is there an issue on this command ?
is there a new command directly by @vysheng that made this function simplest ?

**Comment by raku-cat** [comment]
A feature to make this program not print every message you received since last started is desperately needed, I'm in chats which results in a few thousand messages a day, trying to casually use this program results in having to wait a few minutes, after which it seems to just give up trying to load all the messages and let you use the program normally, but will carry on if you quit and restart it, I'm currently waiting for it to sync every message I've gotten since november of last year.

**Comment by Paulchen232** [comment]
Are you use lua?
